src/libmongoc/src/mongoc/mongoc-optional.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mongoc-prelude.h"
+
+#ifndef MONGOC_OPTIONAL_H
+#define MONGOC_OPTIONAL_H
+
+#include <bson/bson.h>
+
+#include "mongoc-macros.h"
+
+BSON_BEGIN_DECLS
+
+typedef struct {
+    bool value;
+    bool is_set;
+} mongoc_optional_t;
+
+MONGOC_EXPORT (void)
+mongoc_optional_init (mongoc_optional_t *opt);
+
+MONGOC_EXPORT (bool)
+mongoc_optional_set (mongoc_optional_t *opt);
```

Comment:
Nit: `mongoc_optional_set` doesn't sound like a getter. Maybe consider renaming this to `mongoc_optional_is_set`?

Reply:
oh ha I didn't consider that!  Yes, is_set is a much clearer choice.